data-prepper-plugins/common/src/main/java/com/amazon/dataprepper/plugins/buffer/BlockingBuffer.java

```diff
@@ -80,15 +92,29 @@ public BlockingBuffer(final String pipelineName) {
 
     @Override
     public void doWrite(T record, int timeoutInMillis) throws TimeoutException {
+        long nanos = TimeUnit.MILLISECONDS.toNanos(timeoutInMillis);
+        final int c;
         try {
-            boolean isSuccess = blockingQueue.offer(record, timeoutInMillis, TimeUnit.MILLISECONDS);
-            if (!isSuccess) {
-                throw new TimeoutException(format("Pipeline [%s] - Buffer is full, timed out waiting for a slot",
-                        pipelineName));
+            writeLock.lockInterruptibly();
```

Comment:
maybe this line through 106 can go into a function that is named intuitively. this write function is becoming a little unreadable. something like "waitForWriteLockAndBufferCapacity()" (but maybe less wordy)

Reply:
Sure.